src/flypad-backend/src/Aircraft/AircraftProcedures.h

```diff
@@ -16,285 +19,284 @@ struct ProcedureStep {
+  ProcedureStep{"TCAS Switch TA/RA", 4090, false, 1000, "(L:A32NX_SWITCH_TCAS_POSITION) 0 ==",                         "0 (>L:A32NX_SWITCH_TCAS_POSITION)"},
+  ProcedureStep{"WX Radar Mode",     4110, false, 1000, "(L:XMLVAR_A320_WEATHERRADAR_MODE) 1 ==",                      "1 (>L:XMLVAR_A320_WEATHERRADAR_MODE)"},
+  ProcedureStep{"WX Radar Off",      4100, false, 1000, "(L:XMLVAR_A320_WEATHERRADAR_SYS) 1 ==",                       "1 (>L:XMLVAR_A320_WEATHERRADAR_SYS)"},
 };
-// @formatter:on
-
-class AircraftProcedures {
-  vector<ProcedureStep*>* coldAndDark = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* powered = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* readyForPushback = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* readyForTaxi = new vector<ProcedureStep*>;
-  vector<ProcedureStep*>* readyForTakeoff = new vector<ProcedureStep*>;
-  // just to get a list of IDs
-  vector<ProcedureStep*>* all = new vector<ProcedureStep*>;
 
-public:
-  AircraftProcedures() {
-    // Map the procedure groups
-
-#ifdef DEBUG
-    // This is used to generate a list of IDs which can be printed to console to
-    // add them to the EFB code to display the current step.
-    all->insert(all->end(), POWERED_CONFIG_ON->begin(), POWERED_CONFIG_ON->end());
-    all->insert(all->end(), PUSHBACK_CONFIG_ON->begin(), PUSHBACK_CONFIG_ON->end());
-    all->insert(all->end(), TAXI_CONFIG_ON->begin(), TAXI_CONFIG_ON->end());
-    all->insert(all->end(), TAKEOFF_CONFIG_ON->begin(), TAKEOFF_CONFIG_ON->end());
-    all->insert(all->end(), TAKEOFF_CONFIG_OFF->begin(), TAKEOFF_CONFIG_OFF->end());
-    all->insert(all->end(), TAXI_CONFIG_OFF->begin(), TAXI_CONFIG_OFF->end());
-    all->insert(all->end(), PUSHBACK_CONFIG_OFF->begin(), PUSHBACK_CONFIG_OFF->end());
-    all->insert(all->end(), POWERED_CONFIG_OFF->begin(), POWERED_CONFIG_OFF->end());
-
-    std::for_each(all->begin(), all->end(), [&](ProcedureStep* item) {
-      std::cout << item->id << " = " << item->description << std::endl;
+  static constexpr std::size_t COLD_DARK_SIZE = TAKEOFF_CONFIG_OFF.size() + TAXI_CONFIG_OFF.size() + PUSHBACK_CONFIG_OFF.size() + POWERED_CONFIG_OFF.size();
+  static constexpr std::size_t POWERED_SIZE = TAKEOFF_CONFIG_OFF.size() + TAXI_CONFIG_OFF.size() + PUSHBACK_CONFIG_OFF.size() + POWERED_CONFIG_ON.size();
+  static constexpr std::size_t RDY_FOR_PUSHBACK_SIZE = TAKEOFF_CONFIG_OFF.size() + TAXI_CONFIG_OFF.size() + POWERED_CONFIG_ON.size() + PUSHBACK_CONFIG_ON.size();
+  static constexpr std::size_t RDY_FOR_TAXI_SIZE = TAKEOFF_CONFIG_OFF.size() + POWERED_CONFIG_ON.size() + PUSHBACK_CONFIG_ON.size() + TAXI_CONFIG_ON.size();
+  static constexpr std::size_t RDY_FOR_TAKEOFF_SIZE = POWERED_CONFIG_ON.size() + PUSHBACK_CONFIG_ON.size() + TAXI_CONFIG_ON.size() + TAKEOFF_CONFIG_ON.size();
+
+  std::array<const ProcedureStep*, COLD_DARK_SIZE> coldAndDark;
```

Comment:
Rather keep these a vector - it is planned to maybe move the procedure config to a file which is read at runtime.